### PR TITLE
Add futex syscalls

### PR DESCRIPTION
**Pull request by joboet**
Adds the futex syscalls to `hermit-abi`.

### CODE REVIEW

File: hermit-abi/src/lib.rs [hunk 1]
```diff
@@ -101,6 +108,7 @@ pub const LOW_PRIO: Priority = Priority::from(1);
 pub struct Handle(usize);
 
 pub const NSEC_PER_SEC: u64 = 1_000_000_000;
+pub const FUTEX_RELATIVE_TIMEOUT: usize = 1;
```
Comment (mkroening) [inline hermit-abi/src/lib.rs:111]: Should this not match the type of `flags`?

```suggestion
pub const FUTEX_RELATIVE_TIMEOUT: u32 = 1;
```

**Review by mkroening — CHANGES_REQUESTED** [review]
Looks good to me apart from one type. :)

File: hermit-abi/src/lib.rs (hunk 1 above)
Comment (joboet) [reply hermit-abi/src/lib.rs:111]: Oh yes, of course. Thanks!

**Review by mkroening — APPROVED** [review]
bors r+

**Comment by mkroening** [comment]
I'll publish a new version on crates.io tomorrow.

**Comment by mkroening** [comment]
hermit-abi v0.2.6 has been published on crates.io.

**Pull request merged**